state/store.go

```diff
@@ -54,6 +55,14 @@ func abciResponsesKey(height int64) []byte {
 	return encodeKey(prefixABCIResponses, height)
 }
 
+func stateKey() []byte {
+	res, err := orderedcode.Append(nil, prefixState)
```

Comment:
If this is a static key it should be a constant.

Reply:
how would I do that? My understanding is that I can't set a constant to the return value of a function so I thought to just run stateKey() once and set it as a field in the state store. Alternatively I could set it as some constant and hope that it doesn't find itself in the middle of another prefix

Reply:
Just run the function and grab the return value, put it in a constant. Although I suppose `[]byte` can't be a constant, so it'd have to be a variable, but it should be a static global variable. If you'd like to use an orderedcode call to generate it for clarity, just do it in an `init()` function.

Making sure it doesn't overlap with other prefixes has to be done through discipline and good organization/documentation, in the same sense that there's nothing stopping you from using the same key prefix twice. You have the same problem already, since `prefixState` could easily have the same value as some other prefix.

Reply:
Do you think it would be a good idea to but all the prefix keys in the same file. I think first concern is that no keys within state store overlap with each other. Second concern, would be that there is no overlap through out all tm db's

Reply:
Yes, if a package has a bunch of keys spread across many files it's a good idea to locate them in a single place.

I think overlap across databases is less of an issue really, it was probably a bad idea to mention it since I think it's going to be hard to keep track of it. If someone wants to put multiple stores in a single database then they should use a `PrefixDB` or something.